proto/rig/api/v1/metrics/service.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+
+package api.v1.metrics;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "api/v1/metrics/metrics.proto";
+
+service Service {
+  // Get a topic by name and tags.
+  // The tags must not contain wildcards, and thus a 1-1 match is required.
+  rpc GetTopic(GetTopicRequest) returns (GetTopicResponse) {}
+  // Query topics by name and tags returning all matching topics.
+  rpc QueryTopics(QueryTopicsRequest) returns (QueryTopicsResponse) {}
+
+  // Read metrics for a topic.
+  rpc ReadMetrics(ReadMetricsRequest) returns (ReadMetricsResponse) {}
+  // Read metrics for many topics.
+  rpc ReadMetricsMany(ReadMetricsManyRequest)
+      returns (ReadMetricsManyResponse) {}
+
+  // Aggregate metrics for a topic, bucketed by time.
+  rpc BucketedMetrics(BucketedMetricsRequest)
+      returns (BucketedMetricsResponse) {}
+  // Aggregate metrics for many topics, bucketed by time.
+  rpc BucketedMetricsMany(BucketedMetricsManyRequest)
+      returns (BucketedMetricsManyResponse) {}
```

Comment:
Maybe we shouldn't expose topics in the platform API but instead make it project/env/capsule specific. 
I.e.
```
GetMetricsByCapsule(projectID, environmentID, capsuleID string, from ,to time.Time, agg Aggregation) ([]Metric, error)
```
As the public API. Then we do the topic lookup stuff internally. The reason is then we handle lookups with rbac entirely within the platform with some hardcoded logic as to how we map a project,env,capsule triple to topics tags.

Reply:
```
rpc GetMetrics(GetMetricsRequest) GetMetricsResponse

message GetMetricsRequest {
  string project = 1;
  string env = 2;
  string capsule = 3;
  string metric_type = 4;
  Timestamp from = 5;
  Timestamp to = 6;
  Aggregation aggregation = 7;
  Duration bucket_size = 8;
}

message GetMetricsResponse {
  repeated Metric metrics = 1;
}
```

Reply:
Then we can do validation, e.g. `metric_type` must be given, but `project, env,capsule` are optional (and thus wildcardable)

Reply:
yes makes sense

Reply:
or wouldn't it make sense, to get an idea of what metrics exist before starting to fetch and process them?